src/deviation.rs

```diff
@@ -0,0 +1,378 @@
+
+    /// Computes the [L1 distance] between `self` and `other`.
+    ///
+    /// ```text
+    ///  n
+    ///  ∑  |aᵢ - bᵢ|
+    /// i=1
+    /// ```
+    ///
+    /// where `self` is `a` and `other` is `b`.
+    ///
+    /// The following **errors** may be returned:
+    ///
+    /// * `MultiInputError::EmptyInput` if `self` is empty
+    /// * `ShapeMismatch` if `self` and `other` don't have the same shape
+    ///
+    /// [L1 distance]: https://en.wikipedia.org/wiki/Taxicab_geometry
+    fn l1_dist(&self, other: &ArrayBase<S, D>) -> Result<A, MultiInputError>
+    where
+        A: AddAssign + Clone + Signed;
+
+    /// Computes the [L∞ distance] between `self` and `other`.
+    ///
+    /// ```text
+    /// max(|aᵢ - bᵢ|)
+    ///  ᵢ
+    /// ```
+    ///
+    /// where `self` is `a` and `other` is `b`.
+    ///
+    /// The following **errors** may be returned:
+    ///
+    /// * `MultiInputError::EmptyInput` if `self` is empty
+    /// * `ShapeMismatch` if `self` and `other` don't have the same shape
+    ///
+    /// [L∞ distance]: https://en.wikipedia.org/wiki/Chebyshev_distance
+    fn linf_dist(&self, other: &ArrayBase<S, D>) -> Result<A, MultiInputError>
+    where
+        A: Clone + PartialOrd + Signed;
+
+    /// Computes the [mean absolute error] between `self` and `other`.
+    ///
+    /// ```text
+    ///        n
+    /// 1/n *  ∑  |aᵢ - bᵢ|
+    ///       i=1
+    /// ```
+    ///
+    /// where `self` is `a` and `other` is `b`.
+    ///
+    /// The following **errors** may be returned:
+    ///
+    /// * `MultiInputError::EmptyInput` if `self` is empty
+    /// * `ShapeMismatch` if `self` and `other` don't have the same shape
+    ///
+    /// **Panics** if the type cast from `A` to `f64` fails.
+    ///
+    /// [mean absolute error]: https://en.wikipedia.org/wiki/Mean_absolute_error
+    fn mean_abs_err(&self, other: &ArrayBase<S, D>) -> Result<f64, MultiInputError>
+    where
+        A: AddAssign + Clone + Signed + ToPrimitive;
+
+    /// Computes the [mean squared error] between `self` and `other`.
+    ///
+    /// ```text
+    ///        n
+    /// 1/n *  ∑  |aᵢ - bᵢ|²
+    ///       i=1
+    /// ```
+    ///
+    /// where `self` is `a` and `other` is `b`.
+    ///
+    /// The following **errors** may be returned:
+    ///
+    /// * `MultiInputError::EmptyInput` if `self` is empty
+    /// * `ShapeMismatch` if `self` and `other` don't have the same shape
+    ///
+    /// **Panics** if the type cast from `A` to `f64` fails.
+    ///
+    /// [mean squared error]: https://en.wikipedia.org/wiki/Mean_squared_error
+    fn mean_sq_err(&self, other: &ArrayBase<S, D>) -> Result<f64, MultiInputError>
+    where
+        A: AddAssign + Clone + Signed + ToPrimitive;
+
+    /// Computes the unnormalized [root-mean-square error] between `self` and `other`.
+    ///
+    /// ```text
+    /// √ mse(a, b)
+    /// ```
+    ///
+    /// where `self` is `a` and `other` is `b`.
```

Comment:
Can we do a single sentence?
```
where `self` is `a`, `other` is `b` and `mse` is the mean-squared-error.
```

Reply:
Oh yes. Sorry this is a copy and paste error. Thanks for spotting.